### PR TITLE
Added a dump packet function

### PR DESCRIPTION
**Pull request by TheXDS**
Sligtly better detection loop
Help text tabbed

### CODE REVIEW

File: scan.c [hunk 1]
```diff
@@ -97,11 +102,17 @@ int corsairlink_device_scanner(libusb_context *context)
...
 					} else {
+						msg_debug("No (device_id 0x%02X)\n", device_id);
+						
 						corsairlink_handle_close(scanlist[scanlist_count].handle);
 						continue;
 					}
 				}
-			}
+				else
+				{
```
Comment (audiohacked) [inline scan.c]: Curly brackets should be on the same line as the else.

File: print.c [hunk 2]
```diff
@@ -42,3 +44,14 @@ int print(enum msglevel level, const char *fmt, ...)
 	}
 	return 0;
 }
+
+void dump_packet(uint8_t *packet, int size)
```
Comment (audiohacked) [inline print.c]: Function definitions should return 'int' for error codes, even if the function can't possibly produce an error.

File: print.h [hunk 3]
```diff
@@ -31,6 +31,7 @@ enum msglevel {
 extern int verbose;
 
 int print(enum msglevel level, const char *fmt, ...) __attribute__((format(printf, 2, 3)));
+void dump_packet(uint8_t *packet,int size);
```
Comment (audiohacked) [inline print.h]: Again, should return an 'int'.

File: scan.c (hunk 1 above)
Comment (TheXDS) [reply scan.c]: hmmm... I guess vscode misaligned that...

File: print.h (hunk 3 above)
Comment (TheXDS) [reply print.h]: Ok, sure... I'm gonna change it...

File: protocol/hid/core.c [hunk 4]
```diff
@@ -26,32 +26,31 @@
...
-
-	ii = 1;
-	// Read Device ID: 0x3b = H80i. 0x3c = H100i. 0x41 = H110i. 0x42 = H110i Extreme
-	commands[ii++] = CommandId++; // Command ID
-	commands[ii++] = ReadOneByte; // Command Opcode
-	commands[ii++] = DeviceID; // Command data...
-	commands[ii++] = 0x00;
+	uint8_t ii = 0;
 
+	commands[++ii] = CommandId++; // Command ID
```
Comment (TheXDS) [inline protocol/hid/core.c]: I had to do this change because the command was being malformed. my unit seems to be detected fine now.

**Comment by audiohacked** [comment]
Only thing I would suggest is appending the cli option to the end of the list instead of editing long_options and inserting the option in the middle of the list. Other than that I'm going to merge your changes. Any minor formatting cleanup will be done later.

**Comment by TheXDS** [comment]
Awesome! Sure, I will take that in mind next time...

**Pull request merged**